File: employee/src/main/java/com/mtc/employee/services/EmployeeService.java
package com.mtc.employee.services;

import com.mtc.employee.models.Employee;
import com.mtc.employee.repositories.EmployeeRepository;
import org.springframework.scheduling.annotation.Async;
import org.springframework.stereotype.Service;

import java.util.Collection;
import java.util.Optional;
import java.util.Set;

@Service
public class EmployeeService {

    //Injection field injection
    private final EmployeeRepository employeeRepository;

    //Constructor injection
    public EmployeeService(EmployeeRepository employeeRepository) {
        this.employeeRepository = employeeRepository;
    }

    //GET ALL EMPLOYEES
    public Collection<Employee> getEmployees(){
        return employeeRepository.findAll();
    }

    //GET A SINGLE EMPLOYEE - RETURN OPTIONAL
    public Optional<Employee> getEmployee(final Long id){
        return employeeRepository.findById(id);
    }

    //ADD EMPLOYEE
    public Employee addEmployee(final Employee employee){
        return employeeRepository.save(employee);
    }

    //UPDATE EMPLOYEE
    public Employee updateEmployee(final Long id, final Employee employee){
        Optional<Employee> employeeOptional = employeeRepository.findById(id);
        if (employeeOptional.isPresent()){
            Employee oldEmployee = employeeOptional.get();
            //oldEmployee = employee;
            oldEmployee.setEmpID(employee.getEmpID());
            oldEmployee.setFirstName(employee.getFirstName());
            oldEmployee.setMiddleName(employee.getMiddleName());
            oldEmployee.setLastName(employee.getLastName());
            oldEmployee.setIdNumber(employee.getIdNumber());
            oldEmployee.setPosition(employee.getPosition());
            oldEmployee.setSalary(employee.getSalary());

            return employeeRepository.save(oldEmployee);
        }

        return null;
    }


    //DELETE EMPLOYEE
    public void deleteEmployee(Employee employee){
        employeeRepository.delete(employee);
    }



}

File: employeeManagement/src/app/services/employee.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class EmployeeService {

  baseUrl:string = "http://localhost:8080/employees/"
  constructor(private http: HttpClient) { }

  //Get All Employees
  getEmployees(): Observable<any>{
    return this.http.get(this.baseUrl)
  }

  //Get a single Employee
  getEmployee(id:number): Observable<any>{
    return this.http.get(this.baseUrl+id)
  }

  //Add Employee
  addEmployee(e): Observable<any>{
    return this.http.post(this.baseUrl, e)
  }

  //Update
  updateEmployee(id, e): Observable<any>{
    return this.http.put(this.baseUrl+id, e)
  }

  //Delete
  deleteEmployee(id): Observable<any>{
    return this.http.delete(this.baseUrl+id)
  }

}
